src/main.rs

```diff
@@ -41,8 +38,22 @@ fn main() -> Result<()> {
             // `EDITOR`/`--editor` is mandatory if action is `--edit`
             let editor = &opts.editor.unwrap();
             ragenix::edit(&rule, &identities, editor, &mut std::io::stdout())?;
-        } else if opts.rekey {
-            ragenix::rekey(&rules, &identities, &mut std::io::stdout())?;
+        } else if let Some(paths) = opts.rekey {
+            if paths.is_empty() {
+                // Option passed but no files specified - rekey all
+                ragenix::rekey(&rules, &identities, true, &mut std::io::stdout())?;
+            } else {
+                let paths_normalized = paths
+                    .into_iter()
+                    .map(util::canonicalize_rule_path)
+                    .collect::<Result<Vec<PathBuf>>>()?;
+                let chosen_rules = rules
+                    .into_iter()
+                    .filter(|x| paths_normalized.contains(&x.path))
+                    .collect::<Vec<ragenix::RagenixRule>>();
+
+                ragenix::rekey(&chosen_rules, &identities, false, &mut std::io::stdout())?;
+            }
```

Comment:
I think it would be better if you did this in the `ragenix` module. Since you are already adding a new parameter to `rekey`, it could be the given paths. Alternatively, adding a new `rekey_{some,given,chosen}` function is also fine for me.